### PR TITLE
Alternate ACTNARS generic approach

### PR DESCRIPTION
**Pull request by alistairjevans**
Alternate approach for https://github.com/autofac/Autofac/pull/1099.

### CODE REVIEW

File: test/Autofac.Test/Features/ResolveAnything/ResolveAnythingTests.cs [hunk 1]
```diff
@@ -257,6 +256,16 @@ private static IContainer CreateResolveAnythingContainer()
             return cb.Build();
         }
 
+        public class GenericType<T>
```
Comment (tillig) [inline test/Autofac.Test/Features/ResolveAnything/ResolveAnythingTests.cs:259]: Where is this getting used in the tests?

Comment (alistairjevans) [reply test/Autofac.Test/Features/ResolveAnything/ResolveAnythingTests.cs:259]: It's not, my bad; left over from when I looked at bringing in the tests from the original PR. Removed.

**Comment by alistairjevans** [comment]
Are we considering these breaking changes because of the predicate removal?

**Comment by tillig** [comment]
No, the predicate is still running on the type before it gets to the generic part, so it still works. The tests are passing, too, so I'm unsure whether having it in the generic type parameter part ever really mattered. I feel like this is sort of 0.0.1 "bug fix" stuff right now.

**Comment by alistairjevans** [comment]
Ok, cool, this is ready to go then. I'll just link the issue.

**Pull request merged**